lib/createTopLevelExpect.js

```diff
@@ -224,9 +224,18 @@ function createExpectIt(expect, expectations, forwardedFlags) {
             groupEvaluation.every(({ promise }) => promise.isFulfilled())
           )
         ) {
-          expect.fail(output => {
-            writeGroupEvaluationsToOutput(output, groupEvaluations);
-          });
+          if (
+            groupEvaluations.length === 1 &&
+            groupEvaluations[0].length === 1
```

Comment:
Is that the bit that identifies an immediate `expect.it()`? I confess that at first glance I don’t follow how we infer that.

Reply:
Yeah, it detects that we've just executed `expect.it(...)(...)` and not eg. `expect.it(...).and(...)(...)`

Reply:
Would you be willing to add a comment about that? I can foresee some head scratching about this check later and that might be a cheap way to defuse it.

Reply:
Sure, but I don't really want to merge this as-is.

Reply:
Haha got it! I confess the thought did cross my mind about how else we could identify this case

Reply:
I think we need to go back to the drawing board and find out how to allow lifting out a diff from an `expect.it` error.